emily/handler/src/api/models/deposit/responses.rs

```diff
@@ -30,15 +30,47 @@ pub struct GetDepositsResponse {
 #[serde(rename_all = "camelCase")]
 pub struct UpdateDepositsResponse {
     /// Deposit infos: deposits with a little less data.
-    pub deposits: Vec<DepositWithStatus>,
+    pub deposits: Vec<DepositWithStatusSchemed>,
+}
+
+impl<T> From<T> for UpdateDepositsResponse
+where
+    T: IntoIterator<Item = DepositWithStatus>,
+{
+    fn from(value: T) -> Self {
+        Self {
+            deposits: value.into_iter().map(Into::into).collect(),
+        }
+    }
 }
 
 /// Wrapper for deposit with status code. Used for multi-status responses.
-#[derive(Clone, Default, Debug, PartialEq, Hash, Serialize, Deserialize, ToSchema, ToResponse)]
+#[derive(Clone, Debug, PartialEq, Hash, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct DepositWithStatus {
-    /// The fully extracted and validated deposit request.
-    pub deposit: Deposit,
+    /// The fully extracted and validated deposit request, or
+    /// an error message.
+    pub deposit: Result<Deposit, String>,
     /// HTTP status code, returned as part of multi-status responses.
     pub status: u16,
 }
+
+/// Workaround to make utopia generate openapi. Used only as last step before sending
+/// UpdateDepositsResponce
+#[derive(Clone, Default, Debug, PartialEq, Hash, Serialize, Deserialize, ToSchema, ToResponse)]
+#[serde(rename_all = "camelCase")]
+pub struct DepositWithStatusSchemed {
+    deposit: Option<Deposit>,
+    error: Option<String>,
+    status: u16,
+}
```

Comment:
Hmmm, this type seems like the one that we should use. Although it is weaker, in the sense that the `Result<_, _>` conveys the information that we want, this the `DepositWithStatus` struct is used just as a response, so I'd document the expected condition that exactly one of the `deposit` field or the `error` field will be `None`.

So, I think this struct should be the actual `DepositWithStatus`. I'd remove the existing `DepositWithStatus` and rename this one as `DepositWithStatus`. And we'll need to update the existing comments and add a comment to the new error field.

Reply:
Oh and same thing for withdrawals.